### PR TITLE
NC13 is required

### PR DESCRIPTION
**Pull request by enoch85**

### CODE REVIEW

**Comment by enoch85** [comment]
@morph027 @Ark74 Please continue the discussion here. 

Any progress @morph027?

**Comment by morph027** [comment]
prepared a local copy, will do some polishing and push it tomorrow.

**Comment by enoch85** [comment]
@morph027 As you can see I did some changes. Hope it's not to hard to merge them in to your branch.

**Comment by morph027** [comment]
Thanks, will have a look. But if we're using the ReadonlyREST plugin, we don't need the reverse proxy config anymore ;) Will dig through anyway.

**Comment by enoch85** [comment]
Great! 

Want to verify the downloaded file with SHA1. What am I missing here?

```
echo "Downloading readonlyrest..."
rm -f "/tmp/readonlyrest-1.16.15_es$ES_VERSION.zip"
wget -q -T 10 -t 2 "https://github.com/nextcloud/vm/raw/master/apps/fulltextsearch-files/readonlyrest-1.16.15_es$ES_VERSION.zip" -P /tmp
mkdir -p "$GPGDIR"
wget -q -T 10 -t 2 "https://raw.githubusercontent.com/nextcloud/vm/master/apps/fulltextsearch-files/readonlyrest-1.16.15_es$ES_VERSION.zip.sha1" -P "$GPGDIR"
chmod -R 600 "$GPGDIR"

echo "Verifying checksums..."
if [ $(cat "$GPGDIR/readonlyrest-1.16.15_es$ES_VERSION.zip.sha1") ] == $(sha1sum "/tmp/readonlyrest-1.16.15_es$ES_VERSION.zip" | awk '{print $1}')
then
    echo "Checksum OK!"
else
msg_box "Checksum was not OK.

Please report this to $ISSUES."
fi
```

**Comment by morph027** [comment]
Make the sha1 file look like this:

```
9d3dfe321788dfaa90fbf28a7c41f92486da22a9  readonlyrest-1.16.15_es6.1.1.zip
```
Then run:

```
sha1sum -c readonlyrest-1.16.15_es6.1.1.zip.sha1 
```

This will give you:

```
readonlyrest-1.16.15_es6.1.1.zip: OK
```

**Comment by morph027** [comment]
```bash
cd /tmp
if sha1sum "readonlyrest-1.16.15_es$ES_VERSION.zip.sha"; then
    echo "Checksum OK!"
    cd -
else
msg_box "Checksum was not OK.
Please report this to $ISSUES."
exit 1
fi
```

**Comment by enoch85** [comment]
@morph027 Nice one, did another one before I saw your comment. Feel free to change.

File: apps/fulltextsearch.sh [hunk 1]
```diff
@@ -83,37 +76,66 @@ then
...
+rm -rf "$GPGDIR"
+rm -f /tmp/fulltextsearch-files/readonlyrest-1.16.15_es"$ES_VERSION".zip
+exit 1
+fi
+
 if [ -d /usr/share/elasticsearch ]
 then
     cd /usr/share/elasticsearch/bin
-    check_command ./elasticsearch-plugin install "$APP"/fulltextsearch-files/readonlyrest-1.16.15_es"$ES_VERSION".zip
+    check_command ./elasticsearch-plugin file:///tmp/fulltextsearch-files/readonlyrest-1.16.15_es"$ES_VERSION".zip
```
Comment (Ark74) [inline apps/fulltextsearch.sh]: Here is missing `install`
`check_command ./elasticsearch-plugin install file:///tmp/fulltextsearch-files/readonlyrest-1.16.15_es"$ES_VERSION".zip`

**Comment by enoch85** [comment]
@daita Everytihing seems to be working now in current state except `sudo -u www-data php /var/www/nextcloud/occ fulltextsearch:index`

What seems to be the issue?

**Comment by enoch85** [comment]
YML might be wrong...

**Comment by enoch85** [comment]
@morph027 Any progress on your end?

**Comment by enoch85** [comment]
@morph027 @daita @Ark74 Right now it fails with:

```
+ occ_command files_fulltextsearch:configure '{"files_pdf":"1","files_office":"1"}'
+ check_command sudo -u www-data php /var/www/nextcloud/occ files_fulltextsearch:configure '{"files_pdf":"1","files_office":"1"}'
+ sudo -u www-data php /var/www/nextcloud/occ files_fulltextsearch:configure '{"files_pdf":"1","files_office":"1"}'
{
    "files_local": "1",
    "files_external": "0",
    "files_encrypted": "0",
    "files_federated": "0",
    "files_size": "20",
    "files_pdf": "1",
    "files_office": "1",
    "files_image": "0",
    "files_audio": "0"
}
+ occ_command fulltextsearch:index
+ check_command sudo -u www-data php /var/www/nextcloud/occ fulltextsearch:index
+ sudo -u www-data php /var/www/nextcloud/occ fulltextsearch:index
indexing Files.

                                                               
  [Elasticsearch\Common\Exceptions\NoNodesAvailableException]  
  No alive nodes found in your cluster                         
                                                               

fulltextsearch:index [--output [OUTPUT]]

+ printf '\e[0;91mSorry but something went wrong. Please report this issue to https://github.com/nextcloud/vm/issues and include the output of the error message. Thank you!\e[0m\n'
Sorry but something went wrong. Please report this issue to https://github.com/nextcloud/vm/issues and include the output of the error message. Thank you!
+ echo 'sudo -u www-data php /var/www/nextcloud/occ fulltextsearch:index failed'
sudo -u www-data php /var/www/nextcloud/occ fulltextsearch:index failed
+ exit 1

```

Does the service have to be restated before you can index?

**Comment by morph027** [comment]
It has to be up and running. You can check elasticsearch log at `/var/log/elasticsearch/elasticsearch.log` or have a look at http://localhost:9200 at the vm.

**Comment by enoch85** [comment]
@morph027 Some more output:

```
daniel@ubuntu:~$ sudo curl http://localhost:9200
{"error":{"root_cause":[{"type":"status_exception","reason":"forbidden","header":{"WWW-Authenticate":"Basic"}}],"type":"status_exception","reason":"forbidden","header":{"WWW-Authenticate":"Basic"}},"status":401}
```

**Comment by enoch85** [comment]
@morph027 I will stop my editing so that you can update your local copy. I just wanted to fix the occ_command stuff. Now when that's in place it's only the full text search stuff left, and basically, get the `index` command to work.

**Comment by enoch85** [comment]
@morph027 Are you working on this or not? 

Since the scripts are fetched everytime we can concentrate on fixing the last details in `fulltextsearch.sh` - it will be updated in real time if we merge to master. I want to merge this as I'm done with the changes I wanted to make.

**Comment by morph027** [comment]
Uh, sorry, was on a short trip the last days. Also i'm quite busy at work today, but will try to have a look at the changes this evening. Only problem i see so far is that ES is not running. I'll check this out.

**Comment by enoch85** [comment]
@morph027 Sorry to nag you, but any progress?

**Comment by morph027** [comment]
Uh, sorry, release week at work ;)

For me, it worked, ES was running and the script was able to configure the plugin. I don't have any changes to the branch right now.

**Comment by enoch85** [comment]
@morph027 I totally understand. :)

Hmm, so not even an error with the `occ_command fulltextsearch:index`?

**Pull request merged**